src/roiextractors/extractors/hdf5imagingextractor/hdf5imagingextractor.py

```diff
@@ -96,20 +102,29 @@ def __init__(
     def __del__(self):
         self._file.close()
 
-    @check_get_frames_args
-    def get_frames(self, frame_idxs, channel=0):
-        if frame_idxs.size > 1 and np.all(np.diff(frame_idxs) > 0) or frame_idxs.size == 1:
-            return self._video[channel, frame_idxs]
-            # return lazy_ops.DatasetView(self._video).lazy_slice[channel, frame_idxs]
+    def get_frames(self, frame_idxs: ArrayType, channel: Optional[int] = 0):
+
+        # Fancy indexing is non performant for h5.py with long frame lists
+        if frame_idxs is not None:
+            slice_start = np.min(frame_idxs)
+            slice_stop = min(np.max(frame_idxs) + 1, self.get_num_frames())
+        else:
+            slice_start = 0
+            slice_stop = self.get_num_frames()
+
+        if self._data_has_channels_axis and channel is not None:
+            frames = self._standard_video.lazy_slice[slice_start:slice_stop, :, :, channel]
+        elif self._data_has_channels_axis and channel is None:
+            frames = self._standard_video.lazy_slice[slice_start:slice_stop, ...]
+        elif not self.data_has_channel_axis and channel is None:
+            frames = self._standard_video.lazy_slice[slice_start:slice_stop, ...][:, np.newaxis]
```

Comment:
This is a lot of detail just for handling a case that isn't often seen. Is this tested in any way right now?

Reply:
Should either (a) add some tests or (b) add a `NotImplementedError` for when `channel is not None` since that isn't something the rest of the package really supports yet.

Reply:
Personally favoring option (b) since we have no examples anywhere of multi-channel data. We could build tests for HDF5 and numpy extractors with synthetic data, but it's not really an important feature and we have other higher priority things across the ecosystem right now.

Reply:
Done.